spark3/src/main/java/org/apache/iceberg/spark/source/IcebergSource.java

```diff
@@ -56,48 +61,62 @@ public boolean supportsExternalMetadata() {
     String path = options.get("path");
-
-    if (path.contains("/")) {
-      HadoopTables tables = new HadoopTables(conf);
-      return tables.load(path);
+    List<String> ident;
+    try {
+      ident = scala.collection.JavaConverters.seqAsJavaList(SparkSession.active().sessionState().sqlParser().parseMultipartIdentifier(path));
+    } catch (ParseException e) {
+      try {
+        ident = scala.collection.JavaConverters.seqAsJavaList(SparkSession.active().sessionState().sqlParser().parseMultipartIdentifier(String.format("`%s`", path)));
+      } catch (ParseException ignored) {
+        throw new RuntimeException(e);
+      }
+    }
+    if (ident.size() == 1) {
+      return Pair.of(null, TableIdentifier.of(defaultNamespace, ident.get(0)));
+    } else if (ident.size() == 2) {
+      if (catalogManager.isCatalogRegistered(ident.get(0))) {
+        return Pair.of(ident.get(0), TableIdentifier.of(defaultNamespace, ident.get(1))); //todo what if path?
+      } else {
+        return Pair.of(null, TableIdentifier.of(ident.toArray(new String[0])));
+      }
     } else {
-      HiveCatalog hiveCatalog = HiveCatalogs.loadCatalog(conf);
-      TableIdentifier tableIdentifier = TableIdentifier.parse(path);
-      return hiveCatalog.loadTable(tableIdentifier);
+      if (catalogManager.isCatalogRegistered(ident.get(0))) {
+        return Pair.of(ident.get(0), TableIdentifier.of(ident.subList(1, ident.size()).toArray(new String[0])));
+      } else {
+        return Pair.of(null, TableIdentifier.of(ident.toArray(new String[0])));
+      }
     }
   }
 
-  private Table getTableAndResolveHadoopConfiguration(Map<String, String> options, Configuration conf) {
-    // Overwrite configurations from the Spark Context with configurations from the options.
-    mergeIcebergHadoopConfs(conf, options);
-
-    Table table = findTable(options, conf);
-
-    // Set confs from table properties
-    mergeIcebergHadoopConfs(conf, table.properties());
-
-    // Re-overwrite values set in options and table properties but were not in the environment.
-    mergeIcebergHadoopConfs(conf, options);
-
-    return table;
+  @Override
+  public Identifier extractIdentifier(CaseInsensitiveStringMap options) {
+    TableIdentifier tableIdentifier = tableIdentifier(options).second();
+    return Identifier.of(tableIdentifier.namespace().levels(), tableIdentifier.name());
   }
 
-  private static void mergeIcebergHadoopConfs(Configuration baseConf, Map<String, String> options) {
-    options.keySet().stream()
-        .filter(key -> key.startsWith("hadoop."))
-        .forEach(key -> baseConf.set(key.replaceFirst("hadoop.", ""), options.get(key)));
+  @Override
+  public String extractCatalog(CaseInsensitiveStringMap options) {
+    String catalogName = tableIdentifier(options).first();
+    return (catalogName == null) ? SupportsCatalogOptions.super.extractCatalog(options) : catalogName;
```

Comment:
The default implementation always returns `spark_catalog`, not the current catalog. Since we want to use the current catalog when it isn't defined in the identifier, `tableIdentifier(options)` should fill it in. That would simplify this logic because it should always be non-null.

Reply:
Ahh, yes. I just re-read the comments. Make sense, fixed.